server/graphql/types/src/types/purchase_order.rs

```diff
@@ -0,0 +1,220 @@
+        let loader = ctx.get_loader::<DataLoader<NameByIdLoader>>();
+
+        let result = match &self.row().donor_link_id {
+            Some(id) => loader
+                .load_one(NameByIdLoaderInput::new(&store_id, id))
+                .await?
+                .map(NameNode::from_domain),
+            None => None,
+        };
+
+        Ok(result)
+    }
+    pub async fn reference(&self) -> &str {
+        &self.row().reference
+    }
+    pub async fn currency_id(&self) -> &Option<String> {
+        &self.row().currency_id
+    }
+    pub async fn foreign_exchange_rate(&self) -> &Option<f64> {
+        &self.row().foreign_exchange_rate
+    }
+    pub async fn shipping_method(&self) -> &Option<String> {
+        &self.row().shipping_method
+    }
+    pub async fn sent_datetime(&self) -> &Option<NaiveDateTime> {
+        &self.row().sent_datetime
+    }
+    pub async fn contract_signed_datetime(&self) -> &Option<NaiveDateTime> {
+        &self.row().contract_signed_datetime
+    }
+    pub async fn advance_paid_datetime(&self) -> &Option<NaiveDateTime> {
+        &self.row().advance_paid_datetime
+    }
+    pub async fn received_at_port_datetime(&self) -> &Option<NaiveDate> {
+        &self.row().received_at_port_datetime
+    }
+    pub async fn expected_delivery_datetime(&self) -> &Option<NaiveDate> {
+        &self.row().expected_delivery_datetime
+    }
+    pub async fn supplier_agent(&self) -> &Option<String> {
+        &self.row().supplier_agent
+    }
+    pub async fn authorising_officer_1(&self) -> &Option<String> {
+        &self.row().authorising_officer_1
+    }
+    pub async fn authorising_officer_2(&self) -> &Option<String> {
+        &self.row().authorising_officer_2
+    }
+    pub async fn additional_instructions(&self) -> &Option<String> {
+        &self.row().additional_instructions
+    }
+    pub async fn heading_message(&self) -> &Option<String> {
+        &self.row().heading_message
+    }
+    pub async fn agent_commission(&self) -> &Option<f64> {
+        &self.row().agent_commission
+    }
+    pub async fn document_charge(&self) -> &Option<f64> {
+        &self.row().document_charge
+    }
+    pub async fn communications_charge(&self) -> &Option<f64> {
+        &self.row().communications_charge
+    }
+    pub async fn insurance_charge(&self) -> &Option<f64> {
+        &self.row().insurance_charge
+    }
+    pub async fn freight_charge(&self) -> &Option<f64> {
+        &self.row().freight_charge
+    }
+    pub async fn freight_conditions(&self) -> &Option<String> {
+        &self.row().freight_conditions
+    }
+
+    pub async fn lines(&self, ctx: &Context<'_>) -> Result<PurchaseOrderLineConnector> {
+        let loader = ctx.get_loader::<DataLoader<PurchaseOrderLinesByPurchaseOrderIdLoader>>();
+        let result_option = loader.load_one(self.row().id.clone()).await?;
+
+        let result = result_option.unwrap_or(vec![]);
+        Ok(PurchaseOrderLineConnector::from_vec(result))
+    }
```

Comment:
I wasn't quite sure why - but existing pattern is to use loader here for querying lines rather than mapping directly from service layer even if the service layer contains the item line.

You can see this for example in stock_line (also invoice_line and requisition_line) where the struct contains an itemRow, but we still use a loader in graphql to take the id from that struct, and then load through a loader.

Is there some optimisation this allows that I am not aware of?

I wasn't sure - but copied this pattern in anycase.